### PR TITLE
Code Quality: Introduced IStorageArchiveService

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Summary

I'd like to provide single and simple way of compression and decompression.
In this effort, I'd also like to use `ILocatableStorable` to identify compression folder or decompression destination instead of `string` path in the future.

As a sidenote, I removed all null-ref warnings in modified classes (#9895).

### Resolved / Related Issues

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Related #14572

### Steps used to test these changes

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. ✅ Open Files app
2. ✅ Do `Extract files`
3. ✅ Do `Extract here (smart)`
4. ✅ Do `Extract her`
5. ✅ Do `Extract to 'folder name'`
6. ✅ Do `Create archive`
7. ✅ Do `Create .zip`
8. ✅ Do `Create .7z`

> Ensured Status Center has an item for each, and decompressed/compressed file/folder name is increaded as `Decompressed files (2)`.

### CODE REVIEW

File: src/Files.App/Actions/Content/Archives/Compress/CompressIntoArchiveAction.cs [hunk 1]
```diff
@@ -24,7 +24,22 @@ public override async Task ExecuteAsync(object? parameter = null)
...
+			{
+				// Get the current directory path
+				directory = context.ShellPage.FilesystemViewModel.WorkingDirectory.Normalize();
+
+				// Get the library save folder if the folder is library item
+				if (App.LibraryManager.TryGetLibrary(directory, out var library) && !library.IsEmpty)
+					directory = library.DefaultSaveFolder;
+
+				// Gets the file name from the directory path
+				fileName = SystemIO.Path.GetFileName(sources.Length is 1 ? sources[0] : directory);
```
Comment (yaira2) [inline src/Files.App/Actions/Content/Archives/Compress/CompressIntoArchiveAction.cs:41]: Can this code be shared across the various actions?

Comment (0x5bfa) [reply src/Files.App/Actions/Content/Archives/Compress/CompressIntoArchiveAction.cs:41]: Sure, I can put this in the service.

Comment (0x5bfa) [reply src/Files.App/Actions/Content/Archives/Compress/CompressIntoArchiveAction.cs]: Done

**Review by yaira2 — APPROVED** [review]

**Review by hishitetsu — APPROVED** [review]
It works as expected. LGTM

**Pull request merged**